Business Finder/NearbyBusinessesTableViewController.m

```diff
@@ -23,7 +25,30 @@ - (void)viewDidLoad {
         self.delegate.nearbyBusinessesTableViewController = self;
         self.delegate.dataSource = [NearbyBusinessesDataSource new];
     }
+    self.refreshControl = [UIRefreshControl new];
+    [self.refreshControl addTarget:self.delegate
+                            action:@selector(updateBusinesses)
+                  forControlEvents:UIControlEventValueChanged];
+}
+
+-(void)endRefreshing{
+    // We use `performSelector:withObject:afterDelay` because it schedules the selector so that its run the next time
+    // through the current run loop in the default mode. If we just call the selector directly,
+    // the run loop will sometimes be in tracking mode and ignore our request.
+    [self.refreshControl performSelector:@selector(endRefreshing) withObject:nil afterDelay:0.0];
+}
+
+- (void)viewDidAppear:(BOOL)animated {
+    [self.refreshControl beginRefreshing];
+    CGFloat offset = self.tableView.contentOffset.y-self.refreshControl.frame.size.height;
+    [self.tableView setContentOffset:CGPointMake(0, offset) animated:YES];
     [self.delegate startInitialLoad];
 }
 
+-(void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    MapViewController *mapViewController = segue.destinationViewController;
+    BusinessCell *cell = sender;
+    mapViewController.business = cell.business;
```

Comment:
I've modified the code to basically do this, but instead of directly accessing an array, I delegate retrieval through `NearbyBusinessesTVCDelegate`, which in turn delegates to `NearbyBusinessesDataSource`.